modules/caddyhttp/fileserver/staticfiles.go

```diff
@@ -388,17 +395,26 @@ func (fsrv *FileServer) ServeHTTP(w http.ResponseWriter, r *http.Request, next c
 		// of transparent; however we do need to set the Etag:
 		// https://caddy.community/t/gzipped-sidecar-file-wrong-same-etag/16793
 		if etag == "" {
-			etag = calculateEtag(compressedInfo)
+			etag = calculateEtag(info)
 		}
 
 		break
 	}
 
 	// no precompressed file found, use the actual file
 	if file == nil {
-		fsrv.logger.Debug("opening file", zap.String("filename", filename))
-
```

Comment:
I think we should preserve the debug log, it's helpful for tracing. In fact, maybe we should add more debug logs in here relating to which file is being picked depending on the precompressed options.